nixos/modules/services/hardware/libinput.nix

```diff
@@ -249,47 +250,38 @@ in {
     ];
 
   options = {
-
     services.libinput = {
-      enable = mkEnableOption "libinput" // {
-        default = config.services.xserver.enable;
+      enable = mkEnableOption ''libinput.
+        ::: {.note}
+        The mouse and touchpad suboptions only work on X11.
+        :::
+      '' // {
+        default = xcfg.enable;
         defaultText = lib.literalExpression "config.services.xserver.enable";
       };
       mouse = mkConfigForDevice "mouse";
       touchpad = mkConfigForDevice "touchpad";
     };
   };
 
-
   config = mkIf cfg.enable {
+    services.xserver.modules = lib.mkIf xcfg.enable [ pkgs.xorg.xf86inputlibinput ];
 
-    services.xserver.modules = [ pkgs.xorg.xf86inputlibinput ];
-
-    environment.systemPackages = [ pkgs.xorg.xf86inputlibinput ];
+    environment.systemPackages = lib.mkIf xcfg.enable [ pkgs.xorg.xf86inputlibinput ];
 
     environment.etc =
       let cfgPath = "X11/xorg.conf.d/40-libinput.conf";
-      in {
+      in lib.mkIf xcfg.enable {
         ${cfgPath} = {
           source = pkgs.xorg.xf86inputlibinput.out + "/share/" + cfgPath;
         };
       };
 
     services.udev.packages = [ pkgs.libinput.out ];
 
-    services.xserver.inputClassSections = [
+    services.xserver.inputClassSections = lib.mkIf xcfg.enable [
```

Comment:
I can't follow you.

Reply:
I meant that `udev` enablement and XServer config should be in separate modules since they are orthogonal, as mentioned in https://github.com/NixOS/nixpkgs/pull/304902#issuecomment-2092923845

Reply:
We just moved everything into one place from multiple sources to make things easier. I just used a mkMerge instead.

Reply:
`libinput` used to be disabled by default. It is not needed for regular mouse, only for trackpads. Can we revert to old behaviour?